lib/gooddata/models/user_filters/user_filter_builder.rb

```diff
@@ -307,7 +309,13 @@ def self.create_user_profile_mapping(filters, project_users, options = {})
             found_list[login] = user.links['self'] if user
           end
         else
-          domain_users = domain.users
+          if @all_domain_users[domain.name].nil?
```

Comment:
please log splunk when domain.users is called, we should log missing_list.size, domain.name, domain_users.size, use_cache=true/false (@all_domain_users[domain.name] != null)

Reply:
Added logs. Thank you